data/transactions/transaction.go

```diff
@@ -38,9 +38,9 @@ func (txid Txid) String() string {
 	return fmt.Sprintf("%v", crypto.Digest(txid))
 }
 
-// UnmarshalText initializes the Address from an array of bytes.
-func (txid *Txid) UnmarshalText(text []byte) error {
```

Comment:
I had to rename `Txid.UnmarshalText` because it led to asymmetrical treatment in encoding and decoding.

Prior to this PR, JSON encoding a `Txid` object would result in a base64-encoded byte array, but our decoder would use this `UnmarshalText` method, which only works for a base32-encoded string.

I fixed this by renaming the method so now decoding will work from a base64-encoded byte array.

Reply:
No, the encoder must not look at the `.String()` method.  This produces base64 bytes:
```go
var txid transactions.Txid
bytes := protocol.EncodeJSONStrict(&txid)
fmt.Printf("txid bytes: %s", string(bytes)) // produces "txid bytes: \"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=\""
```

Reply:
does not look like this: if not special marshal methods then it array is encoded as json array:

> Marshal traverses the value v recursively. If an encountered value implements [Marshaler](https://pkg.go.dev/encoding/json@go1.23.0#Marshaler) and is not a nil pointer, Marshal calls [Marshaler.MarshalJSON] to produce JSON. If no [Marshaler.MarshalJSON] method is present but the value implements [encoding.TextMarshaler](https://pkg.go.dev/encoding#TextMarshaler) instead, Marshal calls [encoding.TextMarshaler.MarshalText](https://pkg.go.dev/encoding#TextMarshaler.MarshalText) and encodes the result as a JSON string. The nil pointer exception is not strictly necessary but mimics a similar, necessary exception in the behavior of [Unmarshaler.UnmarshalJSON].
> Array and slice values encode as JSON arrays, except that []byte encodes as a base64-encoded string, and a nil slice encodes as the null JSON value.

It also does not match what Jason mentioned:
>JSON encoding a Txid object would result in a base64-encoded byte array,

Something to look into I guess

Reply:
ah, it is codec, not `encoding/json`, ok.